src/apis/crypto-compare.ts

```diff
@@ -26,13 +26,27 @@ export async function fetchExchangeRate(
   includeUSDRate?: boolean,
 ): Promise<{
   conversionDate: number;
-  conversionRate: number;
-  usdConversionRate: number;
+  conversionRate: number | null;
+  usdConversionRate: number | null;
 }> {
+  if (!currency || !nativeCurrency) {
```

Comment:
Suggestion: It might be simpler to move this condition into the currency rate controller, and never call `fetchExchangeRate` if either of these are null. That would allow simplifying this type signature, and it'd reduce some duplication of the "fallback" values for the conversion rates (they're already initialized to `null` in the controller). Semantically it also seems better that way, as there's not really any "fetching" happening in this case.

Reply:
Agreed as well!